### PR TITLE
display the total clicks and total impressions in advertising pages

### PR DESCRIPTION
**Pull request by therocket-gr**
Related to #

## Proposed Changes

2638-gh-tumblr/a8c-dsp

## Why are these changes being made?
we want to add a total impressions and total clicks (for all campaigns) for a user in the advertising pages (both in ready to promote and campaigns page)

the added elements look like this (ss from browser)
Screenshot 2024-10-15 at 18 15 32

and ss from figma 
Screenshot 2024-10-15 at 18 17 11

please note that I haven't added the icons for more info (the ones with the 'i') but I am missing the context of what will happen if hovered ( I guess we need some kind of popover with some additional messages)

## Update:
I added a function to format large numbers properly..

as we have users who might have very large numbers of campaigns and their impressions (maybe also clicks) can be added to have millions of results 

now for example 100000 will be displayed as 100K 125500 will be displayed as 125.5K 

1000000 will be displayed as 1M 

1255000 will be displayed as 1.255M and so on (for millions I added 3 decimals to have some better understanding of the amount)

to test this you can just hardcode the line 

`totalClicks={ campaignsStats?.total_clicks }`

to
`totalClicks={ 1500000 }`
in

client/my-sites/promote-post-i2/main.tsx

 
*

## Testing Instructions

please check out the branch (or test the preview links) check that the elements appear as expected
(I checked the layout both when we do have a banner and when we don't.. you can also test by hardcoding this line
`const showBanner = ! campaignsIsLoading && ( totalCampaignsUnfiltered || 0 ) < 3;` and change it to showBanner = true or false accordingly

test that the layout looks ok in tablet and mobile..

do a CR.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/promote-post-i2/components/campaigns-total-stats/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+				<div className="campaigns-total-stats__item">
+					<div className="campaigns-total-stats__label">
+						{ translate( 'Total people reached' ) }
+					</div>
+					<div className="campaigns-total-stats__result">
+						{ formatNumber( totalImpressions || 0 ) }
+					</div>
+				</div>
+				<div className="campaigns-total-stats__item">
+					<div className="campaigns-total-stats__label">{ translate( 'Total visitors' ) }</div>
```
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaigns-total-stats/index.tsx]: We are using clicks instead of visitors here. The designs are not fully updated. 

We wanted to use clicks to differentiate with the visitors metric. Not all clicks end up being visitors.

File: config/development.json [hunk 2]
```diff
@@ -20,8 +20,8 @@
 	"login_url": "https://wordpress.com/wp-login.php",
 	"logout_url": "https://wordpress.com/wp-login.php?action=logout&redirect_to=https%3A%2F%2F|subdomain|wordpress.com",
 	"facebook_api_key": "249643311490",
-	"dsp_stripe_pub_key": "pk_live_51LYYzQF53KN4RFN0oHFYHpzmHGpuUAXZ6ygDAFxM4XkRArRsET0orrofmsynkd9DhFaOGEsDZ3RC4f8mgI0zhEyN000X8i0Mqx",
-	"dsp_widget_js_src": "https://dsp.wp.com/widget.js",
+	"dsp_stripe_pub_key": "pk_test_51LYYzQF53KN4RFN0iNDEzgRCLRC461l8K3m30wCger66JBFAzgU5f6o1qIIKLLGSZUtzKe4wGJFkU2x4Jn9GRATu00StWVfEDc",
```
Comment (sbarbosa) [inline config/development.json]: I think you forgot to revert this change. We need to keep the live version of the key and widget URL in here.

**Review by sbarbosa — COMMENTED** [review]
Hi @therocket-gr! 
I found a couple of things that we need to change, but overall, the changes look great and worked as expected.

File: config/development.json (hunk 2 above)
Comment (therocket-gr) [reply config/development.json]: nice catch.. sorry for that.. it is clearly a mistake

Comment (therocket-gr) [reply config/development.json]: fixed

File: client/my-sites/promote-post-i2/components/campaigns-total-stats/index.tsx (hunk 1 above)
Comment (therocket-gr) [reply client/my-sites/promote-post-i2/components/campaigns-total-stats/index.tsx]: fixed.. thanks

**Review by sbarbosa — APPROVED** [review]
LGTM 👍

**Comment by jjolmo** [comment]
image
LGTM

**Review by jjolmo — APPROVED** [review]

File: client/my-sites/promote-post-i2/utils/index.ts [hunk 3]
```diff
@@ -244,6 +244,15 @@ export const formatNumber = ( number: number, onlyPositives = false ): string =>
 	return number.toLocaleString();
 };
 
+export const formatLargeNumber = ( number: number ): string => {
```
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/utils/index.ts:247]: I searched in Calypso, and I found this function that do a similar thing:
https://github.com/Automattic/wp-calypso/tree/trunk/client/lib/format-number-compact

Do you think we could use that one instead? It might behave differently, which is what you wanted to achieve here.

Comment (therocket-gr) [reply client/my-sites/promote-post-i2/utils/index.ts:247]: the function is a bit strange.. it wont format millions properly. i tested and I think my method works better. i just added a small improvement to add locale formatting in the number (so for some countries that use comma instead of a point for decimals it will display the result better)

**Review by sbarbosa — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16912196

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @therocket-gr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16912196).
